docs/rpc_api_versioning.md

```diff
@@ -0,0 +1,272 @@
+ 
+ - The `V1` version of the endpoint, which is now the older version, should be 
+ explicitly accessible via `existsaddressv1`.
+
+ - The default method name, `existsaddress` will resolve to the current version 
+ of the endpoint which is `existsaddressv2` in this case.
+
+ This scheme will be enforced via the JSON-RPC type and handler registrations. 
+
+<a name="Example" />
+
+### 4. Versioned Enpoint Example
+
+Taking into consideration the accessibility scheme outlined above, implementing 
+the `V2` version of the `existaddress` JSON-RPC endpoint will comprise of the 
+following components: 
+
+```go
+     // JSON-RPC versioned endpoint command definition.
+
+    // ExistsAddressV2Cmd defines the existsaddressv2 JSON-RPC command.
+    type ExistsAddressV2Cmd struct {
+        Address       string
+        PaymentScript string
+    }
+    // ...
+```
+```go
+    // ExistsAddress JSON-RPC type registrations.
+ 	dcrjson.MustRegister(Method("existsaddress"), (*ExistsAddressV2Cmd)(nil), flags)
+	dcrjson.MustRegister(Method("existsaddressv1"), (*ExistsAddressCmd)(nil), flags)
+	dcrjson.MustRegister(Method("existsaddressv2"), (*ExistsAddressV2Cmd)(nil), flags)
+    // ...
+```
+
+```go
+    // ExistsAddressV2 JSON-RPC Handler definition.
+
+    // handleExistsAddress implements the existsaddress command.
+    //
+    // Deprecated: This will be removed on the next versioned endpoint
+    // implementation.
+    func handleExistsAddress(_ context.Context, s *Server, cmd interface{}) (interface{}, error) {
+        if s.cfg.ExistsAddresser == nil {
+            return nil, rpcInternalError("Exists address index disabled",
+                "Configuration")
+        }
+
+        c := cmd.(*types.ExistsAddressCmd)
+        // ...
+    }
+
+    // handleExistsAddressV2 implements the existsaddressv2 command.
+    func handleExistsAddressV2(_ context.Context, s *Server, cmd interface{}) (interface{}, error) {
+        if s.cfg.ExistsAddresser == nil {
+            return nil, rpcInternalError("Exists address index disabled",
+                "Configuration")
+        }
+
+        c := cmd.(*types.ExistsAddressV2Cmd)
+        // ...
+    }
+
+``` 
+
+```go
+    // ExistsAddress JSON-RPC handler registrations.
+    var rpcHandlersBeforeInit = map[types.Method]commandHandler{
+        // ...
+        "existsaddress":         handleExistsAddressV2,
+        "existsaddressv1":       handleExistsAddress,
+        "existsaddressv2":       handleExistsAddressV2,
+        // ...
+    }
+```
+
+```go
+// ExistsAddress JSON-RPC result type registrations.
+var rpcResultTypes = map[types.Method][]interface{}{
+    // ...
+	"existsaddress":         {(*bool)(nil)},
+	"existsaddressv1":       {(*bool)(nil)},
+	"existsaddressv2":       {(*bool)(nil)},
+    // ...
+}
+```
+
+<a name="DeprecationAndRemovalScheme" />
+
+### 5. Versioned Enpoint Deprecation And Removal Scheme
+
+Only two versioned endpoints will be maintained per JSON-RPC endpoint. The 
```

Comment:
"Each JSON-RPC method will only have two maintained endpoint versions. The introduction of a new endpoint version should immediately indicate that the oldest maintained endpoint will be deprecated. This is expected...."